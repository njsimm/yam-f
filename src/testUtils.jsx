import React from "react";
import { render } from "@testing-library/react";
import UserContext from "./utils/UserContext";
import SidebarContext from "./utils/SidebarContext";
import { vi } from "vitest"; // Import vi from vitest

// Mock user data
export const mockUser = {
  id: 1,
  email: "demo@example.com",
  username: "demoUser",
  password: "password123",
  first_name: "Demo",
  last_name: "User",
  is_admin: false,
  date_created: "2024-06-24T12:00:00.000Z",
};

// Mock sidebar context data
export const mockSidebarContext = {
  isSidebarOpen: true,
  toggleSidebar: vi.fn(),
};

// UserProvider component to wrap components with UserContext
const UserProvider = ({ children, currentUser = mockUser }) => {
  return (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
  );
};

// SidebarProvider component to wrap components with SidebarContext
const SidebarProvider = ({ children, sidebarContext = mockSidebarContext }) => {
  return (
    <SidebarContext.Provider value={sidebarContext}>
      {children}
    </SidebarContext.Provider>
  );
};

const AllProviders = ({
  children,
  currentUser = mockUser,
  sidebarContext = mockSidebarContext,
}) => {
  return (
    <UserProvider currentUser={currentUser}>
      <SidebarProvider sidebarContext={sidebarContext}>
        {children}
      </SidebarProvider>
    </UserProvider>
  );
};

export * from "@testing-library/react";
export { UserProvider, SidebarProvider, AllProviders };
