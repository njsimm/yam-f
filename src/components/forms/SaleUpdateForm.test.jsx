import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { AllProviders } from "../../testUtils";
import SalesUpdateForm from "./SaleUpdateForm";
import { expect, test, vi } from "vitest";

// Mock updateSale function
const mockUpdateSale = vi.fn();

vi.mock("react-router-dom", async () => {
  const actual = await vi.importActual("react-router-dom");
  return {
    ...actual,
    useNavigate: () => vi.fn(),
  };
});

test("renders SalesUpdateForm component", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <AllProviders>
        <SalesUpdateForm updateSale={mockUpdateSale} />
      </AllProviders>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
