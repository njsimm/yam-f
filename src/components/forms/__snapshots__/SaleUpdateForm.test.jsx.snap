// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders SalesUpdateForm component 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ya6j6j-MuiPaper-root"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </main>
</DocumentFragment>
`;
